easybuild/easyconfigs/p/Python/Python-3.11.5-custom-ctypes.patch

```diff
@@ -0,0 +1,52 @@
+diff -ruN Python-3.11.5.orig/Lib/ctypes/__init__.py Python-3.11.5/Lib/ctypes/__init__.py
+--- Python-3.11.5.orig/Lib/ctypes/__init__.py	2025-06-03 17:24:09.817225787 +0200
++++ Python-3.11.5/Lib/ctypes/__init__.py	2025-06-03 17:26:02.851166381 +0200
+@@ -13,6 +13,8 @@
+ from _ctypes import ArgumentError
+ 
+ from struct import calcsize as _calcsize
++from ctypes import util
++import re
+ 
+ if __version__ != _ctypes_version:
+     raise Exception("Version number mismatch", __version__, _ctypes_version)
+@@ -349,6 +351,11 @@
+             flags |= _FUNCFLAG_USE_ERRNO
+         if use_last_error:
+             flags |= _FUNCFLAG_USE_LASTERROR
++        if _os.name == "posix":
++            if name and name.endswith(".so"):
++                s = re.sub(r'lib', '', name)
++                s = re.sub(r'\..*', '', s)
++                self._name=util._findLib_ld(s) 
+         if _sys.platform.startswith("aix"):
+             """When the name contains ".a(" and ends with ")",
+                e.g., "libFOO.a(libFOO.so)" - this is taken to be an
+diff -ruN Python-3.11.5.orig/Lib/ctypes/util.py Python-3.11.5/Lib/ctypes/util.py
+--- Python-3.11.5.orig/Lib/ctypes/util.py	2025-06-03 17:24:09.816225753 +0200
++++ Python-3.11.5/Lib/ctypes/util.py	2025-06-03 17:28:52.212009753 +0200
+@@ -209,7 +209,7 @@
+             expr = os.fsencode(expr)
+ 
+             try:
+-                proc = subprocess.Popen(('/sbin/ldconfig', '-r'),
++                proc = subprocess.Popen((os.environ.get('EPREFIX') + '/sbin/ldconfig', '-r'),
```

Comment:
I think this is another reason why we should resolve things at the EasyBlock level: there, we can query if EasyBuild was configured with a `sysroot` build option. If so, we get _that_ value, instead of `EPREFIX` (which I believe is set by Gentoo-prefix, and thus specific to that OS?). That makes this more generic. It also excludes any possible name collisions (someone could set `EPREFIX` as an environment for a totally different purpose - unrelated to Gentoo-Prefix, but `build_option('sysroot')` can _only_ mean one thing: someone wanted to set a sysroot).